api/v1alpha1/thanosqueryfrontend_types.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ThanosQueryFrontendSpec defines the desired state of ThanosQueryFrontend
+type ThanosQueryFrontendSpec struct {
+	CommonThanosFields `json:",inline"`
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default=1
+	Replicas int32 `json:"replicas,omitempty"`
+	// CompressResponses enables response compression
+	// +kubebuilder:default=true
+	CompressResponses bool `json:"compressResponses,omitempty"`
```

Comment:
nit: But i probably wouldnt even bother exposing this for now, we could stick with the defaults. If you want to keep it here might want to make it a pointer and optional?

Reply:
Let's keep it, I think I remember some issue where people were debugging with this.